plugin.program.mceremote/mceremote.py

```diff
@@ -0,0 +1,733 @@
+    except Exception as e:
+        dialog.ok("MCERemote", "Unexpected error writing the ReportMappingTable data: " + e.args[0])
+    except:
+        dialog.ok("MCERemote", "Unexpected error writing the ReportMappingTable data")
+
+# Finished!
+
+    winreg.CloseKey(hkey)
+    dialog.ok("MCERemote", "Done! Please restart Windows to activate the changes.")
+
+
+# **********************************************************************
+# ConfigureButtonSettings
+# -----------------------
+# Open the dialog to configure the settings
+# **********************************************************************
+def ConfigureButtonSettings():
+
+# Open the settings dialog
+
+    _settings.openSettings()
+
+# Ask the user if they want to apply these settings
+
+    if xbmcgui.Dialog().yesno("MCERemote", "Do you want to apply these settings to the remote now?"):
+        ApplyCurrentSettings(False)
+
+
+# **********************************************************************
+# RestoreDefaultButtonSettings
+# ----------------------------
+# Restore the default settings
+# **********************************************************************
+def RestoreDefaultButtonSettings():
+
+# Require confirmation from the user
+
+    dialog = xbmcgui.Dialog()
+    if not dialog.yesno("MCERemote", "This will reset the remote button settings to the Media Center defaults. Do you want to proceed?"):
+        return
+
+# Restore the default settings from the info in _buttonData
+
+    for nextbutton in _buttonData:
+        _settings.setSetting(nextbutton[0], nextbutton[2])
+
+# Ask the user if they want to apply these settings
+
+    if xbmcgui.Dialog().yesno("MCERemote", "Do you want to apply these settings to the remote now?"):
+        ApplyCurrentSettings(False)
+
+
+# **********************************************************************
+# EditKeyboardDotXML
+# --------------------
+# Edit the keyboard.xml in the user's userdata\keymaps folder.
+# **********************************************************************
+def EditKeyboardDotXML():
+
+    import os
+    import subprocess
+
+    dialog = xbmcgui.Dialog()
+
+# Check if we are updating KeyMapEdit.exe
+
+    doupdate = _settings.getSetting("update_keyedit")
+    if doupdate == "true":
+        if dialog.yesno("MCERemote", "Do you want to update the keymap editor from Sourceforge?"):
+            if GetKeyMapEdit():
+                _settings.setSetting("update_keyedit", "false")
+            else:
+                return
+
+# Check whether keyboard.xml exists
+
+    dstpath = xbmcvfs.translatePath("special://home/userdata/keymaps/keyboard.xml")
+    if not os.path.isfile(dstpath):
+        if dialog.yesno("MCERemote", "You don't currently have a keyboard.xml file. Do you want to create one?"):
+            CreateKeyboardDotXML()
+        else:
+            return
+
+# Select the keymap editor: if KeyMapEdit.exe exists use it, otherwise
+# use Notepad.
+
+    srcpath = _DataDir + "KeyMapEdit.exe"
```

Comment:
What is this executable?

Reply:
This is my keymap editor applet. The same applet that MCERemote has been using for all editions before Matrix for the last ten years.